agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -460,7 +460,8 @@ func (cs *CredentialSpecResource) handleSSMCredentialspecFile(originalCredential
 	if length < 2 {
 		return errors.New("Failed to retrieve taskId from taskArn.")
 	}
-	localCredSpecFilePath := fmt.Sprintf("%s\\ssm_%v_%s", cs.credentialSpecResourceLocation, taskArnSplit[length-1], ssmParam)
+	ssmParamSanitizedFileName := strings.Replace(ssmParam[1], "/", "_")
```

Comment:
Should it be `strings.Replace` or `strings.ReplaceAll ` here?

Reply:
`strings.Replace()` takes a fourth argument that specifies the number of replaces that will happen, but when < 0 or undefined, will have no limit on how many replaces it makes. That being said, `ReplaceAll` would be more immediately understandable.

Reply:
I also called this out in the description, but I was at least thinking that there is a potential here for using the whole SSM param path as part of the file name to hit file name length limits. If we wanted to be extra cautious, we could make this where we call the `filepath.Base()` function to generate that part of the file name.

Reply:
I agree, I think it would be great if we could use the whole SSM param path and ensure that it does not hit the file name length limits.

Reply:
actually, having done some quick research, windows [MAX_PATH ](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd) (the whole path including filename) must be less than 260 characters. According to [AWS Docs](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetParametersByPath.html#API_GetParametersByPath_RequestSyntax), max path length for an SSM parameter is 2048 characters. Given that we already take up a decent number of characters from the `credentialSpecResourceLocation` and the `taskId`, I fear we would see a ton of issues using the whole SSM parameter path as part of the filename. So I think in this instance we should revert this to using the `filepath.Base()` function to be safer.

That being said, there is still a possibility that someone can use a very long name for their SSM parameter that doesnt have sub path hierarchies, and that could still break the limit here. If we wanted to be _really_ safe, we would truncate length once the full path is constructed.

Let me know how you want to proceed here and I'll make the appropriate changes.

Reply:
While, we can document the known limitations. It would be neat if we could come up with a way to support both long names and nested sub-path hierarchies.

Reply:
yeah, though i dont think we can do that without first having [a specific windows setting enabled](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd#enable-long-paths-in-windows-10-version-1607-and-later)

Reply:
@rue-nsilverman We could fix this issue by introducing a hierarchy to the local `credential-spec` file alongside volume paths. 

Currently `credential-specs` are modeled as a task resource and all of them are created together. When this task-resource is created, we could plumb the task\container details down to this resource and introduce a hierarchy similar to `CredSpecRoot\task\container\credspec` and wire this to the container appropriately.